components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java

```diff
@@ -359,17 +364,20 @@ private void processResponseMessage(MessageContext synCtx, ConfigurationContext
         CachableResponse response = (CachableResponse) synCtx.getProperty(CachingConstants.CACHED_OBJECT);
 
         if (response != null) {
-            boolean toCache;
+            boolean toCache = true;
             if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(response.getProtocolType())) {
-                String statusCode = msgCtx.getProperty(NhttpConstants.HTTP_SC).toString();
-                // Now create matcher object.
-                Matcher m = response.getResponseCodePattern().matcher(statusCode);
-                if (m.matches()) {
-                    toCache = true;
-                    response.setStatusCode(statusCode);
-                    response.setStatusReason(msgCtx.getProperty(PassThroughConstants.HTTP_SC_DESC).toString());
-                } else {
-                    toCache = false;
+                String statusCode = (String) msgCtx.getProperty(NhttpConstants.HTTP_SC);
+
+                if (statusCode != null) {
+                    // Now create matcher object.
+                    Matcher m = response.getResponseCodePattern().matcher(statusCode);
+                    if (m.matches()) {
+                        toCache = true;
+                        response.setStatusCode(statusCode);
+                        response.setStatusReason((String) msgCtx.getProperty(PassThroughConstants.HTTP_SC_DESC));
+                    } else {
+                        toCache = false;
```

Comment:
Can't we avoid this else block if we have the initial toCache initialized to false, and make it to true only if m.matches() ?

Reply:
I can't initialize it to null because `toCache` has to be true when `statusCode` is null

Reply:
Ok, if so toCache=true at L375 is redundant right?